src/main/webapp/app/exercises/text/participate/text-editor.component.ts

```diff
@@ -174,14 +173,14 @@ export class TextEditorComponent implements OnInit, OnDestroy, ComponentCanDeact
      * @private
      */
     private setupComponentWithInputValues() {
-        if (this.inputExercise) {
-            this.textExercise = this.inputExercise;
+        if (this.inputExercise() !== undefined) {
```

Comment:
tbh. i'm a bit confused here, can `this.inputExercise()` even evaluate to undefined, if you define it as ` input<TextExercise>()` ?

Reply:
If the input was not set than it could be undefined.